### PR TITLE
Feat: MatchMaking Screen

### PR DESCRIPTION
**Pull request by privilegemendes**
## Description
This PR introduces the match making screen for users to wait until they are assigned a room to start a game match

## Changes
- added an event emitter for the server to notify the client if players join or leave the queue
- progress bar to provide a visual indication of available players in the queue
- added a character indicator to provide players a visual indication of which characters have been assigned and how many slots are needed to start a game

### CODE REVIEW

File: src/server/api/routers/lobby.ts [hunk 1]
```diff
@@ -23,10 +23,16 @@ export const lobbyRouter = createTRPCRouter({
         ee.off("queueUpdate", handleEvent);
 
         const { id } = ctx.session.user;
-
         const index = lobbyQueue.indexOf(id);
-        if (index < 0) return;
-        lobbyQueue.splice(index, 1);
+
+        // If the user is in the queue, and they unsubscribe (disconnect), remove them from the queue
+        if (index >= 0) {
```
Comment (ramidecodes) [inline src/server/api/routers/lobby.ts:29]: Isn't equal or bigger than zero always returning true?

**Review by ramidecodes — CHANGES_REQUESTED** [review]
The logic in the Lobby router looks a bit dodgy

**Review by Mautjee — APPROVED** [review]
LGTM

File: src/server/api/routers/lobby.ts (hunk 1 above)
Comment (privilegemendes) [reply src/server/api/routers/lobby.ts]: Whoops, forgot to remove the if statement. All good now

**Review by ramidecodes — APPROVED** [review]

**Pull request merged**